test/junit/scala/tools/nsc/backend/jvm/StringConcatTest.scala

```diff
@@ -0,0 +1,70 @@
+package scala.tools.nsc
```

Comment:
It'd be nice to copy this test to dotty.

Reply:
Ported in https://github.com/lampepfl/dotty/pull/3391